src/main/java/pl/polsl/egradebook/controller/TeacherController.java

```diff
@@ -42,10 +60,162 @@ public String viewGradesAndAttendance(Authentication authentication, Model model
         Iterable<Student> classStudents = studentRepository.findAll();
         model.addAttribute("teacher", loggedTeacher);
         model.addAttribute("grades", gradeRepository.findAllByOrderByStudent_User_SurnameAsc());
-        model.addAttribute("attendance", presenceRepository.findAll());
+        model.addAttribute("attendance", presenceRepository.findAllByLesson_Teacher_User_UserID_OrderByDateDesc(loggedTeacher.getUserID()));
         model.addAttribute("students", studentRepository.findAll());
-
+        model.addAttribute("homeUrl",homeUrl);
         return "teacher-view";
     }
 
+    //cases view for the teacher, coppied from student
+    @GetMapping(path = "/cases/{caseID}")
+    @PreAuthorize("hasAuthority('/teacher/cases/{caseID}')")
+    public String selectedCase(@PathVariable("caseID") int caseID, Model model) {
+        Case foundCase = caseRepository.findById(caseID).orElseThrow(() -> new IllegalArgumentException("Invalid id:" + caseID));
+        model.addAttribute("case", foundCase);
+        model.addAttribute("homeUrl",homeUrl);
+        return "case-content-view";
+    }
+
+    //submits changed attendances for selected class
+    //returns attendance-management.html with model attribute submitSuccessful set to true to show the message on page
+    @PostMapping("/attendance/submit")
+    @PreAuthorize("hasAuthority('/teacher/attendance/submit')")
+    public String submitAttendance(ListWrapper listWrapper, Model model,Authentication authentication)
+    {
+        List<Presence> studentAttendance=listWrapper.getList();
+        System.out.println(studentAttendance);
+        for(Presence p: studentAttendance)
+        {
+            presenceRepository.save(p);
+        }
+        String userName=authentication.getName();
+        Teacher loggedTeacher = teacherRepository.findByUser_UserName(userName);
```

Comment:
Dot. w sumie wszystkich Kontrollerów. Tutaj to się powtarza w każdej funkcji. Można zrobić funkcję
private Teacher getTeacherByUserName(String userName), która to bd zawierała